spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java

```diff
@@ -512,8 +512,9 @@ public void setAdviceChain(Advice... adviceChain) {
 		this.adviceChain = Arrays.copyOf(adviceChain, adviceChain.length);
 	}
 
+	@Nullable
 	protected Advice[] getAdviceChain() {
-		return this.adviceChain;
+		return this.adviceChain == null ? null : Arrays.copyOf(this.adviceChain, this.adviceChain.length);
```

Comment:
Right. And this is exactly what I commented above. 
Why do we need to do this?
Is this something SonarQube complains, too?